FEMTree: Fix infinite loop due to 32-bit integer overflow.

nh2 opened the pull request:
The code

    size_t sz = nodeCount();
    for( unsigned int i=0 ; i<sz ; i++ ) nodeHasData[i] = 0;

looped forever when `nodeCount()` is >= 2^32.

mkazhdan commented:
Thanks. I have modified the code and updated the repo. 
Independently, did you enable the BIG_DATA macro in PreProcessor.h?

nh2 commented:
> Thanks. I have modified the code and updated the repo.

Great, thanks!

I've also posted

* [How can I get a warning when comparing unsigned integers of different size in C and C++?](https://stackoverflow.com/questions/76840686/how-can-i-get-a-warning-when-comparing-unsigned-integers-of-different-size-in-c)
* and a [GCC feature request `Add warning flags to check against integer overflow`](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=110933)

to help find such issues.

> Independently, did you enable the BIG_DATA macro in PreProcessor.h?

Yes, but not in `PreProcessor.hs`, instead I'm currently passing it as a `CFLAGS=-DBIG_DATA` environment variable to `make`.

The project would really benefit from a proper build system such as Meson or CMake, vs. commenting in and out code and `Makefile` lines.
